src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs

```diff
@@ -23,16 +23,16 @@ public class IniStreamConfigurationProvider : StreamConfigurationProvider
         /// </summary>
         /// <param name="stream">The stream of INI data.</param>
         /// <returns>The <see cref="IDictionary{String, String}"/> which was read from the stream.</returns>
-        public static IDictionary<string, string> Read(Stream stream)
+        public static IDictionary<string, string?> Read(Stream stream)
         {
-            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var data = new Dictionary<string, string?>(StringComparer.OrdinalIgnoreCase);
             using (var reader = new StreamReader(stream))
             {
                 string sectionPrefix = string.Empty;
 
                 while (reader.Peek() != -1)
                 {
-                    string rawLine = reader.ReadLine();
+                    string rawLine = reader.ReadLine()!;
```

Comment:
(nit) It might be good to add a comment here why this is safe. (I assume it is because Peek returned non `-1`.